filters/auth/tokeninfo_test.go

```diff
@@ -669,16 +676,35 @@ func TestOAuthTokeninfoValidate(t *testing.T) {
 			expectStatus:       http.StatusOK,
 			expectAuthRequests: 0,
 		},
+		{
+			name:               "allow missing token when request host matches opted-out domain",
+			host:               "foo.domain.test",
+			authHeader:         "",
+			expectStatus:       http.StatusOK,
+			expectAuthRequests: 0,
+		},
+		{
+			name:               "allow missing token when request second level host matches opted-out domain",
+			host:               "foo.bar.domain.test",
+			authHeader:         "",
+			expectStatus:       http.StatusOK,
+			expectAuthRequests: 0,
+		},
+		{
+			name:               "allow missing token when request host matches opted-out host exactly",
+			host:               "exact.test",
+			authHeader:         "",
+			expectStatus:       http.StatusOK,
+			expectAuthRequests: 0,
+		},
```

Comment:
Should we test "domain.test" and "not.exact.test" and/or "notexact.test" here as well?